controllers/autoscaling.go

```diff
@@ -257,11 +259,23 @@ func (r *HorizontalRunnerAutoscalerReconciler) calculateReplicasByPercentageRunn
 		scaleDownFactor = sdf
 	}
 
+	selector, err := metav1.LabelSelectorAsSelector(rd.Spec.Selector)
+	if err != nil {
+		return nil, err
+	}
 	// return the list of runners in namespace. Horizontal Runner Autoscaler should only be responsible for scaling resources in its own ns.
 	var runnerList v1alpha1.RunnerList
-	if err := r.List(ctx, &runnerList, client.InNamespace(rd.Namespace)); err != nil {
-		return nil, err
+	if err := r.List(
+		ctx,
+		&runnerList,
+		client.InNamespace(rd.Namespace),
+		client.MatchingLabelsSelector{Selector: selector},
```

Comment:
As I have roughly mentioned in https://github.com/summerwind/actions-runner-controller/pull/355#discussion_r587191728, I thought this might be incorrect.

We don't include `runner-template-hash` into account here, so at least this can result in runners from multiple RunnerReplicaSet to match. It would double `numRunners` while replacing RunnerReplicaSet due to RD.Spec.Template change.

Reply:
Probably we could just use `rd.Spec.Replicas` (that defaults to 1 if missing) instead of `numRunners := len(runnerList.Items)`, while keeping this line as-is.

Reply:
Well, it might be out-of-scope- the above issue should have been there before this PR.